pkg/docker/config/config.go

```diff
@@ -0,0 +1,295 @@
+	// that they are trying to logout of
+	ErrNotLoggedIn = errors.New("not logged in")
+)
+
+// SetAuthentication stores the username and password in the auth.json file
+func SetAuthentication(ctx *types.SystemContext, registry, username, password string) error {
+	return modifyJSON(ctx, func(auths *dockerConfigFile) (bool, error) {
+		if ch, exists := auths.CredHelpers[registry]; exists {
+			return false, setAuthToCredHelper(ch, registry, username, password)
+		}
+
+		creds := base64.StdEncoding.EncodeToString([]byte(username + ":" + password))
+		newCreds := dockerAuthConfig{Auth: creds}
+		auths.AuthConfigs[registry] = newCreds
+		return true, nil
+	})
+}
+
+// GetAuthentication returns the registry credentials stored in
+// either auth.json file or .docker/config.json
+// If an entry is not found empty strings are returned for the username and password
+func GetAuthentication(ctx *types.SystemContext, registry string) (string, string, error) {
+	if ctx != nil && ctx.DockerAuthConfig != nil {
+		return ctx.DockerAuthConfig.Username, ctx.DockerAuthConfig.Password, nil
+	}
+
+	dockerLegacyPath := filepath.Join(homedir.Get(), dockerLegacyCfg)
+	paths := [3]string{getPathToAuth(ctx), filepath.Join(homedir.Get(), dockerCfg, dockerCfgFileName), dockerLegacyPath}
+
+	for _, path := range paths {
+		legacyFormat := path == dockerLegacyPath
+		username, password, err := findAuthentication(registry, path, legacyFormat)
+		if err != nil {
+			return "", "", err
+		}
+		if username != "" && password != "" {
+			return username, password, nil
+		}
+	}
+	return "", "", nil
+}
+
+// GetUserLoggedIn returns the username logged in to registry from either
+// auth.json or XDG_RUNTIME_DIR
+// Used to tell the user if someone is logged in to the registry when logging in
+func GetUserLoggedIn(ctx *types.SystemContext, registry string) string {
+	path := getPathToAuth(ctx)
+	username, _, _ := findAuthentication(registry, path, false)
+	if username != "" {
+		return username
+	}
+	return ""
+}
+
+// RemoveAuthentication deletes the credentials stored in auth.json
+func RemoveAuthentication(ctx *types.SystemContext, registry string) error {
+	return modifyJSON(ctx, func(auths *dockerConfigFile) (bool, error) {
+		// First try cred helpers.
+		if ch, exists := auths.CredHelpers[registry]; exists {
+			return false, deleteAuthFromCredHelper(ch, registry)
+		}
+
+		if _, ok := auths.AuthConfigs[registry]; ok {
+			delete(auths.AuthConfigs, registry)
+		} else if _, ok := auths.AuthConfigs[normalizeRegistry(registry)]; ok {
+			delete(auths.AuthConfigs, normalizeRegistry(registry))
+		} else {
+			return false, ErrNotLoggedIn
+		}
+		return true, nil
+	})
+}
+
+// RemoveAllAuthentication deletes all the credentials stored in auth.json
+func RemoveAllAuthentication(ctx *types.SystemContext) error {
+	return modifyJSON(ctx, func(auths *dockerConfigFile) (bool, error) {
+		auths.CredHelpers = make(map[string]string)
```

Comment:
Hum, this is a bit surprising. Depending on how one looks at it, it either makes perfect sense or it is extremely unexpected.

I guess, why not — @runcom I just want to make sure you have noticed this.

Reply:
THis is for
kpod logout --all

Basically you want to cleanup and logout of your container workflow.  In most cases not really necessary since the default location is in /run/USER/UID, but it is better to be safe and not leave these credentials around.

Reply:
The way `CredHelpers` work for logging in is that the user must set them up in the config file before starting the log-in process; and it seems reasonable that in a (log in; log out; log in) sequence, the credentials should be stored in the same place (i.e. that the users’ configuration should not be thrown away).

OTOH then `kpod logout --all` would either have to ignore the helpers-using domains entirely, or clean everything in the helper, and both can be pretty unexpected (removing items from a server-side shared keyring).

As you say, with `/run` being the default location, it’s likely enough that users of credential helpers will be overriding the path and knowing not to call `logout --all`, and this is at least conservative in not deleting too much.

(I guess I could argue that `logout --all` is a problematic concept in itself, but that’s not going to help, is it?)

Reply:
Another option rather then doing this through the library would be to 
rm -f $autffile 
from 
kpod logout --all.

Reply:
I guess do appreciate that would get rid of me with my pesky questions, but it doesn’t really avoid having to make a decision.

As I said at the very start, “I guess, why not”, all I want here is to make sure this choice has been made deliberately.

Reply:
Not trying to get rid of your pesky questions.  :^) I actually like them because they make me think.  I was just thinking whether or not this call belongs in a library.

Reply:
Just wondering - how is doing rm -f $authfile any different from clearing the `CredHelpers` field in the authfile? At the end of the day you are still getting rid of the `CredHelpers` no?

Reply:
@umohnani8 yup, this is just a matter of whether you want it in the library or not.

Reply:
ah okay. So what is the decision? I don't see a problem with it being in the library just because to me both approaches does the same thing at the end of the day. But you guys know better :)

Reply:
I can live with either design.

@runcom still needs to approve the PR as a whole, so it’s ultimately up to him.

Reply:
@runcom PTAL